addons/website_product_comparison/controllers/main.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+from odoo import http
+from odoo.http import request
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+import json
+
+
+class WebsiteSaleProductComparison(WebsiteSale):
+
+    def get_compute_currency_and_context(self):
+        pricelist_context = dict(request.env.context)
+        pricelist = False
+        if not pricelist_context.get('pricelist'):
+            pricelist = request.website.get_current_pricelist()
+            pricelist_context['pricelist'] = pricelist.id
+        else:
+            pricelist = request.env['product.pricelist'].browse(pricelist_context['pricelist'])
+
+        from_currency = request.env.user.company_id.currency_id
+        to_currency = pricelist.currency_id
+        compute_currency = lambda price: from_currency.compute(price, to_currency)
+        return compute_currency, pricelist_context
+
+    @http.route('/shop/compare/', type='http', auth="public", website=True)
+    def product_compare(self, **post):
+        values = {}
+        if post.get('products'):
+            product_ids = [int(i) for i in post.get('products').split(',') if i.isdigit()]
+            # use search to check read access on each record/ids
+            products = request.env['product.product'].search([('id', 'in', product_ids)]).exists()
+            values['products'] = products
+
+            res = {}
+            for num, product in enumerate(products):
+                for var in product.attribute_line_ids:
+                    cat_name = var.attribute_id.category_id.name
```

Comment:
This can be `False` (e.g. create a new attribute category) and will crash on `spec.split()` in the template